### PR TITLE
Async postgres + proper abstractions + kuick-db (jdbc/h2/postgres + Async postgres) + repository

### PR DESCRIPTION
**Pull request by soywiz**
:)

### CODE REVIEW

File: kuick-db/src/jvmMain/kotlin/kuick/client/postgres/PostgresDriver.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import io.reactiverse.kotlin.pgclient.*
+import io.reactiverse.kotlin.pgclient.PgClient
+import io.reactiverse.pgclient.*
+import io.vertx.core.*
+import kuick.client.db.*
+import kuick.client.sql.*
+
+object PostgresDriver : DbDriver {
+    override suspend fun connect(url: String): DbConnection {
+        val vertx = Vertx.vertx()
```
Comment (maballesteros) [inline kuick-db/src/jvmMain/kotlin/kuick/client/postgres/PostgresDriver.kt]: There should be just one `Vertx.vertx()` instance, and should be global...

File: kuick-db/src/jvmTest/kotlin/kuick/client/repositories/DbModelRepositoryTest.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package kuick.client.repositories
+
+import kotlinx.coroutines.*
+import kuick.client.db.*
+import kuick.client.jdbc.*
+import kuick.repositories.*
+import kotlin.test.*
+
+class DbModelRepositoryTest {
```
Comment (maballesteros) [inline kuick-db/src/jvmTest/kotlin/kuick/client/repositories/DbModelRepositoryTest.kt]: We should have "conformance tests", that is a shared test that takes as an argument a `ModelRepository` implementation. Then, for each implementation, we should just execute the conformance test for that implementation to verify that is "conformant" with the specification.

File: kuick-db/src/jvmMain/kotlin/kuick/client/postgres/PostgresDriver.kt (hunk 1 above)
Comment (soywiz) [reply kuick-db/src/jvmMain/kotlin/kuick/client/postgres/PostgresDriver.kt]: The problem is that if you not close the `Vertx.vertx` instance, the process keeps running. What I can do is to make `PostgresDriver` a class with a companion object where you can pass a `Vertx` instance that is not handled here.

File: kuick-db/src/jvmTest/kotlin/kuick/client/repositories/DbModelRepositoryTest.kt (hunk 2 above)
Comment (soywiz) [reply kuick-db/src/jvmTest/kotlin/kuick/client/repositories/DbModelRepositoryTest.kt]: Agree. But the common serialization strategies are not finished yet. This is just a first step. We should cleanup stuff later.

**Comment by soywiz** [comment]
This is not finished yet, but it is a good cut-off for a first step. It doesn't touch old code (except some default interface implementations), so should be mostly safe.
The new code should be not used just yet, until all the serialization strategies are implemented, and the code is better tested.
Once this is done, squash will be redundant, and we will be able to use the new code, or just update the repositories-squash to use the new common serialization strategies available to all the implementations.

**Pull request merged**